pkg/controllers/user/alert/watcher/workload.go

```diff
@@ -162,11 +181,63 @@ func (w *WorkloadWatcher) checkWorkloadCondition(wl *workload.Workload, alert *v
 		data["workload_kind"] = wl.Kind
 		data["available_percentage"] = strconv.Itoa(percentage)
 		data["available_replicas"] = strconv.Itoa(int(wl.Status.AvailableReplicas))
-		data["desired_replicas"] = strconv.Itoa(int(wl.Status.Replicas))
+		data["desired_replicas"] = strconv.Itoa(int(desiredReplicas))
 
 		if err := w.alertManager.SendAlert(data); err != nil {
 			logrus.Errorf("Failed to send alert: %v", err)
 		}
 	}
 
 }
+
+func (w *WorkloadWatcher) getDesiredReplicas(workloadName string) (int32, error) {
+	var desiredReplicas int32
+	splitted := strings.Split(workloadName, ":")
+	if len(splitted) != 3 {
+		return desiredReplicas, errors.New("invalid workloadName: " + workloadName)
+	}
+	workloadType := strings.ToLower(splitted[0])
+	namespace := splitted[1]
+	name := splitted[2]
+	switch workloadType {
+	case workload.ReplicationControllerType:
+		o, err := w.replicationControllerLister.Get(namespace, name)
+		if err != nil {
+			return desiredReplicas, err
+		}
+		if o.Spec.Replicas != nil {
+			return *o.Spec.Replicas, nil
+		}
+	case workload.ReplicaSetType:
+		o, err := w.replicaSetLister.Get(namespace, name)
+		if err != nil {
+			return desiredReplicas, err
+		}
+		if o.Spec.Replicas != nil {
+			return *o.Spec.Replicas, nil
+		}
+	case workload.DaemonSetType:
+		o, err := w.daemonsetLister.Get(namespace, name)
+		if err != nil {
+			return desiredReplicas, err
+		}
+		return o.Status.DesiredNumberScheduled, nil
+	case workload.StatefulSetType:
+		o, err := w.statefulsetLister.Get(namespace, name)
+		if err != nil {
+			return desiredReplicas, err
+		}
+		if o.Spec.Replicas != nil {
+			return *o.Spec.Replicas, nil
+		}
+	default:
```

Comment:
The switch does not handle job/cronjob, they are considered as deployment kind in those cases.

Reply:
This feature does not include job and cronjob, we return direclty when matching job/cronjob at line 150